src/viking.js

```diff
@@ -1,14 +1,148 @@
 // Soldier
-class Soldier {}
+
+class Soldier {
+  constructor(health, strength) {
+    this.health = health;
+    this.strength = strength;
+  }
+
+  attack () {
+    return this.strength
+  }
+  
+  receiveDamage(damage) {
+    this.damage = damage;
+    this.health = this.health - this.damage;
+  }  
+}
 
 // Viking
-class Viking {}
+
+class Viking extends Soldier {
+  constructor(name, health, strength) {
+    super(health, strength);
+    this.name = name
+  }
+
+  // attack() should be inherited from Soldier, no need to re-implement it
+
+  receiveDamage(damage) {
+    this.health = this.health - damage // nao precisa colocar o this na frente pq o damage só está nesta funcao
+
+    if (this.health > 0) return `${this.name} has received ${damage} points of damage`
+    return `${this.name} has died in act of combat`// nao precisa falar do else pq vai sair do if se for false e cai nesse return
+  }
+
+    battleCry() {
+      return 'Odin Owns You All!'
+    }
+}
 
 // Saxon
-class Saxon {}
 
-// War
-class War {}
+class Saxon extends Soldier { // You don't have to include constructor method since this class will inherit perfectly from the parents class, both, the health and the strength (it extends Soldier class 😉 )
+
+  // attack() should be inherited from Soldier, no need to re-implement it
+
+  receiveDamage(damage) {
+    this.health = this.health - damage
+
+    if (this.health > 0) return `A Saxon has received ${damage} points of damage`
+    return `A Saxon has died in combat`
+  }
+
+}
+
+// War 
+class War {
+  constructor() { 
+    this.vikingArmy = []; // aqui é this pq não estamos criando uma variável, só estamos dizendo que a classe vai ter este atributo
+    this.saxonArmy = []; 
+  }
+  
+  addViking(viking) { // viking aqui é passador
+    this.vikingArmy.push(viking) 
+  }
+
+  addSaxon(saxon) {
+    this.saxonArmy.push(saxon) // não precisa colocar o (i,i) pq diferente da aula não estamos declarando um novo objeto com os atributos i,i 
+  }
+
+ 
+  vikingAttack() {
+    let randomSaxon = this.saxonArmy[Math.floor(Math.random() * this.saxonArmy.length)]; // usamos o let pq senao geraria 2 random diferentes, um pro attack e outro pro if dead
+
+    let randomViking = this.vikingArmy[Math.floor(Math.random() * this.vikingArmy.length)]
+
+    let attack = randomSaxon.receiveDamage(randomViking.strength); 
+        
+    if (randomSaxon.health<=0){
+      this.saxonArmy.splice(indexOf(randomSaxon),1);
```

Comment:
`indexOf` é um método, não uma função!
isto é, ele *precisa acompanhar alguma coisa*!
seria `this.vikingArmy.indexOf(randomViking)`!

E isso ele reclamava no terminal, leia os erros! Ajuda muito!
E o mesmo valo para a função nos saxões!